src/Actions/Blocks.js

```diff
@@ -55,38 +55,34 @@ export const requestPreviousPage = function() {
 export const SET_BLOCK_REQUESTED = `${prefix}/SET_BLOCK_REQUESTED`
 export const ADD_BLOCK_TO_VIEW = `${prefix}/ADD_BLOCK_TO_VIEW`
 export const getBlock = function(number) {
-  return function(dispatch, getState) {
+  return async function(dispatch, getState) {
     let requested = getState().blocks.requested
 
     // If it's already requested, bail
     if (requested[number] === true) {
       return
     }
 
-    let provider = getState().web3.provider
+    let web3Instance = getState().web3.web3Instance
```

Comment:
What's the difference between `provider` and `web3Instance` in this new version of web3? Are they fundamentally the same abstraction?

Reply:
Edit: looks like you described it below.